src/components/layouts/AppMain.vue

```diff
@@ -11,6 +15,16 @@ export default {
     return {
       title: 'Hello! It\'s AppMain'
     }
+  },
+  methods: {
+    toNewMessage () {
+      this.$router.push('/new-message')
+    }
+  },
+  computed: {
+    isNewMessage () {
+      return this.$route.path.includes('new-message') || this.$store.getters['user/isAuth'] === false
```

Comment:
`this.$store.getters['user/isAuth'] === false` -> `!this.$store.getters['user/isAuth'] === false`
`isNewMessage ()` -> `isVisible()`